R/Task_operators.R

```diff
@@ -320,6 +320,18 @@ getTaskData = function(task, subset = NULL, features, target.extra = FALSE, reco
     if (recode.target %nin% c("no", "surv")) {
       res[, tn] = recodeY(res[, tn], type = recode.target, task$task.desc)
     }
+    # first condition checks if 'getTaskData' was called directly, i.e. checks if
+    # the call was from the GlobalEnv
+    # second condition checks which function called 'getTaskData' if call was not
+    # from the GlobalEnv. If cond2 is FALSE, 'getTaskData' was called from
+    # 'subsetTask' in a nested resampling call. In this case we remove x and y
+    # later as we still need it for partitioning
+    if (!identical(parent.frame(n = 1), globalenv()) &&
+        !sys.call(-2) == "subsetTask(.task, .subset)" &&
+        task$task.desc$is.spatial == TRUE) {
+      res$x = NULL
+      res$y = NULL
+    }
   }
```

Comment:
Because I have to make the removal of `x` and `y` dependent on the function which called `getTaskData`.  There are differences between the `train` + `predict` combination and `resample`.  In the latter case, we arrive here (`getTaskData`) twice. The first time we need the complete data, as we need the coordinates for partitioning. The second time, we subset the data to not include `x` and `y` in the training and prediction. 

Here is the updated comment

> In this case we remove x and y later (later = we arrive in 'getTaskData'  twice in a 'resample' call) as we still need it for partitioning in upcoming function calls and only need to remove `x` and `y` before we proceed to the training step. 

More clear now? Happy to expand more if needed.